packages/components/src/InputPhoneNumber/InputMask.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactElement, cloneElement, useState } from "react";
+import styles from "./InputMask.css";
+import { FormFieldProps } from "../FormField";
+
+export interface InputMaskProps {
+  /**
+   * Determines the masking pattern. By default, a `*` is used to be the value
+   * you're expecting the user would type.
+   */
+  readonly pattern: string;
+
+  /**
+   * Change the delimiter when you need to have a `*` as a value that the input
+   * returns.
+   *
+   * @default "*"
+   */
+  readonly delimiter?: string;
+
+  /**
+   * Allow/prevent users adding from more value than the desired pattern.
+   * For example, your pattern could accept 10 characters. If it's strict, your
+   * users can only type 10 characters.
+   *
+   * @default true
+   */
+  readonly strict?: boolean;
```

Comment:
`strict` can mean a lot in this context - consider using something more specific to what this currently is doing, e.g. `limitToPatternLength` ?? naming is hard... `restrictLengthToPattern`? You get the idea.

Reply:
yeah, trying to get a semantic/declarative naming out of it too rather than an imperative one. Something to think about tho 🤔

Reply:
My first instinct would be that "strict" meant it had to conform to ALL validations or else be ... blank/empty probably? 
"Strict" being false might then mean that it tried to format but allowed any input values to be set for the field. 

I wonder if there are different checks that will make up our Mask checks in general: only allow certain characters, limit to the pattern length, only allow the precise number. And then, what to do when they aren't valid: have the field be blank, have the field go to a default or previous value (e.g. date pickers usually default to today or the original input date if they can't parse the input to a date). We might want to have more of a config object (almost like the validations themselves) to allow for flexible specification. 

That's a lot of thoughts beyond what we need right now. But it still makes me think that `strict` isn't quite the prop we want here.

Reply:
i'm thinking it would be private to form field. No one should really be using `InputMask` or `FieldMask` on its own. It's a baked-in component inside the `FormField`

Reply:
Would you be okay if we open this conversation again on the "implement on formfield" task? Or a follow up. I think this component will evolve the more we go through improving it (adding validations, changing the UX a bit)

Reply:
Yup, absolutely. We won't get it right just by pondering ideas, we can iterate as we go! :)

Reply:
yea! I like this kind of comment! makes me really think about what I'm doing